### PR TITLE
Use metastore service in spark

### PR DESCRIPTION
**Pull request by liebstein**

### CODE REVIEW

File: docker/spark/scripts/entrypoint-submit.sh [hunk 1]
```diff
@@ -35,6 +35,16 @@ spark.master $SPARK_MASTER
 spark.ui.port $SPARK_UI_PORT
 " >> /spark/conf/spark-defaults.conf
 
+if [[ ! -z ${HIVE_METASTORE_URI} ]]; then
+echo -e "
+spark.sql.catalogImplementation hive
+spark.sql.hive.convertMetastoreParquet false
+spark.serializer=org.apache.spark.serializer.KryoSerializer
```
Comment (lyogev) [inline docker/spark/scripts/entrypoint-submit.sh]: Why is this related?

File: docker/spark/scripts/entrypoint-submit.sh [hunk 2]
```diff
@@ -35,6 +35,16 @@ spark.master $SPARK_MASTER
 spark.ui.port $SPARK_UI_PORT
 " >> /spark/conf/spark-defaults.conf
 
+if [[ ! -z ${HIVE_METASTORE_URI} ]]; then
+echo -e "
+spark.sql.catalogImplementation hive
+spark.sql.hive.convertMetastoreParquet false
```
Comment (lyogev) [inline docker/spark/scripts/entrypoint-submit.sh]: Why is this related?

File: docker/spark/scripts/entrypoint-submit.sh [hunk 3]
```diff
@@ -35,6 +35,16 @@ spark.master $SPARK_MASTER
 spark.ui.port $SPARK_UI_PORT
 " >> /spark/conf/spark-defaults.conf
 
+if [[ ! -z ${HIVE_METASTORE_URI} ]]; then
+echo -e "
+spark.sql.catalogImplementation hive
```
Comment (lyogev) [inline docker/spark/scripts/entrypoint-submit.sh]: =hive

File: docker/spark/scripts/entrypoint-submit.sh [hunk 4]
```diff
@@ -35,6 +35,16 @@ spark.master $SPARK_MASTER
...
+spark.sql.hive.convertMetastoreParquet false
+spark.serializer=org.apache.spark.serializer.KryoSerializer
+spark.hadoop.hive.metastore.uris thrift://$HIVE_METASTORE_URI
```
Comment (lyogev) [inline docker/spark/scripts/entrypoint-submit.sh]: =

File: e2e/hive/docker-compose.yml [hunk 5]
```diff
@@ -39,7 +42,20 @@ services:
...
+    environment:
+      - CONNECTION_URL=jdbc:mysql://hive-db:3306/hive?useSSL=false
+      - CONNECTION_USER_NAME=root
+      - CONNECTION_PASSWORD=pass
+      - WAREHOUSE_DIR=file:///warehouse
+      - WAIT_HOSTS=hive-db:3306
+    depends_on:
+      - hive-db
+    volumes:
+      - ./output/:/examples/output/
```
Comment (lyogev) [inline e2e/hive/docker-compose.yml]: Not needed, you need this:
- ./warehouse:/warehouse

Comment (liebstein) [reply e2e/hive/docker-compose.yml]: no need for volumes
